### PR TITLE
Drop menu fix for handling closeOnClickInside

### PR DESCRIPTION
**Pull request by sujayjaju**
Fixes implementation of #9131 and includes a fix for #9293

### CODE REVIEW

File: js/foundation.dropdownMenu.js [hunk 1]
```diff
@@ -95,18 +95,23 @@ class DropdownMenu {
...
-
+    
+    // Handle Leaf element Clicks
+    if(_this.options.closeOnClickInside){
+      this.$menuItems.on('click.zf.dropdownmenu touchend.zf.dropdownmenu', function(e) {
+        var hasSub = $elem.hasClass(parClass);
+        if(!hasSub){
+          _this._hide();
+        }
+      }
```
Comment (kball) [inline js/foundation.dropdownMenu.js]: I think you need a closing parens here

Comment (kball) [inline js/foundation.dropdownMenu.js]: Yeah, travis is breaking due to this.

**Review by kball — CHANGES_REQUESTED** [review]
Thanks for taking a pass on this @sujayjaju!  Travis turned up one typo to take a look at, but then we can get on to testing.

**Review by kball — APPROVED** [review]

File: js/foundation.dropdownMenu.js [hunk 2]
```diff
@@ -95,18 +95,23 @@ class DropdownMenu {
...
 
     if (this.options.clickOpen || hasTouch) {
       this.$menuItems.on('click.zf.dropdownmenu touchstart.zf.dropdownmenu', handleClickFn);
     }
-
+    
+    // Handle Leaf element Clicks
+    if(_this.options.closeOnClickInside){
+      this.$menuItems.on('click.zf.dropdownmenu touchend.zf.dropdownmenu', function(e) {
+        var hasSub = $elem.hasClass(parClass);
```
Comment (forkiven) [inline js/foundation.dropdownMenu.js]: Is it just me or is $elem not defined here. Are we missing a  `$elem = $(this)`?

**Review by mikewuu — COMMENTED** [review]
$elem undefined error

File: js/foundation.dropdownMenu.js (hunk 2 above)
Comment (ncoden) [inline js/foundation.dropdownMenu.js]: Yes.

**Comment by kball** [comment]
Yes, this was caught in QA and has been fixed in the 6.3 branch - see: https://github.com/zurb/foundation-sites/pull/9472/files

**Pull request merged**